app/components/verkiezingen/bcsd-voorzitter-alert.js

```diff
@@ -1,130 +1,99 @@
-      this.lastRecomputeTime ===
-        this.installatievergadering.recomputeBCSDNeededTime
-    ) {
-      await timeout(10000);
-      // nothing for now let's try again later
-      this.handleErrorMessage.perform();
-      return;
-    }
-
-    this.lastRecomputeTime =
-      this.installatievergadering.recomputeBCSDNeededTime;
-
-    const voorzitter = await this.getVoorzitterBCSD();
-    const isMemberOfRMW = await this.isMemberOfRMW(voorzitter);
-    const isMemberOfVastBureau = await this.isMemberOfVastBureau(voorzitter);
-
-    this.errorMessage =
-      !voorzitter || isMemberOfRMW || isMemberOfVastBureau
-        ? ''
-        : 'De voorzitter van het BCSD moet lid zijn van de RMW of het Vast Bureau.';
-    await timeout(10000);
-    this.handleErrorMessage.perform();
-  });
-
-  async getVoorzitterBCSD() {
-    return await queryRecord(this.store, 'persoon', {
-      filter: {
-        'is-aangesteld-als': {
-          bekleedt: {
-            bestuursfunctie: {
-              ':uri:': MANDAAT_VOORZITTER_BCSD_CODE,
-            },
-            'bevat-in': {
-              ':uri:': this.bcsdBestuursorgaanInTijd.uri,
-            },
-          },
-        },
-      },
-    });
-  }
-
-  async isMember(persoon, bestuursfunctieCode, hasClassificationFunc) {
-    if (!persoon) {
-      return false;
+  @tracked messages = A();
+
+  isVoorzitterAlsoSchepen = task(async () => {
+    const bcsdMandatarissen = this.args.mandatarissen;
+    const mapping = await Promise.all(
+      bcsdMandatarissen.map(async (mandataris) => {
+        return {
+          mandataris,
+          persoon: await mandataris.isBestuurlijkeAliasVan,
+          isVoorzitter: await (await mandataris.bekleedt).isVoorzitter,
+        };
+      })
+    );
+    const hasVoorzitter = mapping.find(
+      (mapping) => mapping.persoon && mapping.isVoorzitter
+    );
+    if (hasVoorzitter) {
+      const schepen = await this.findMandatarisForOneOfBestuursfunctieCodes(
+        hasVoorzitter.persoon,
+        [MANDAAT_SCHEPEN_CODE_ID, MANDAAT_TOEGEVOEGDE_SCHEPEN_CODE_ID]
+      );
+      if (!schepen) {
+        if (!this.messages.findBy('id', 1)) {
+          this.messages.pushObject({
+            id: 1,
+            message:
+              'Kon geen schepen mandataris vinden voor aangeduide voorzitter.',
+          });
+        }
+      } else {
+        const toRemove = this.messages.findBy('id', 1);
+        if (toRemove) {
+          this.messages.removeObject(toRemove);
+        }
+      }
     }
 
-    const bestuursorgaanInTijd = await findFirst(
-      this.bestuursorganenInTijd,
-      async (bestInTijd) => {
-        return await hasClassificationFunc(bestInTijd);
+    if (hasVoorzitter) {
+      const isLidRMWOrVastBureau =
+        await this.findMandatarisForOneOfBestuursfunctieCodes(
+          hasVoorzitter.persoon,
+          [MANDAAT_LID_RMW_CODE_ID, MANDAAT_LID_VAST_BUREAU_CODE_ID]
+        );
+      if (!isLidRMWOrVastBureau) {
```

Comment:
so this warning can be removed because it is included in the first warning about them having to be schepen

Reply:
For me its a different situation  but if you say it's like that ok 😰